### PR TITLE
Allowed saving timestamp-less messages to Cache (using rospy.Time.now() as their timestamp).

### PR DESCRIPTION
**Pull request by peci1**
Also added a convenience `getLast` method for retrieving the last message in the cache.

### CODE REVIEW

**Comment by tfoote** [comment]
I don't believe we should ever automatically stamp unstamped data. An alternative would be to provide a 2nd argument to which the user could add the timestamp explicitly.

**Comment by peci1** [comment]
Okay, what about adding an `__init__` argument defaulting to False, which, set to True, would allow for automatic timestamping?

**Comment by dirk-thomas** [comment]
That constructor argument sounds like a good option.

File: utilities/message_filters/src/message_filters/__init__.py [hunk 1]
```diff
@@ -121,19 +122,21 @@ def add(self, msg):
     def getInterval(self, from_stamp, to_stamp):
         """Query the current cache content between from_stamp to to_stamp."""
         assert from_stamp <= to_stamp
-        return [m for m in self.cache_msgs
-                if m.header.stamp >= from_stamp and m.header.stamp <= to_stamp]
+        return [self.cache_msgs[i] for i in range(0, len(self.cache_msgs))
+                if self.cache_times[i] >= from_stamp and self.cache_times[i] <= to_stamp]
```
Comment (dirk-thomas) [inline utilities/message_filters/src/message_filters/__init__.py]: Please keep the list comprehension as simple as possible, e.g.:

```
[ m for i, m in enumerate(self.cache_msgs) if self.cache_times[i] >= from_stamp and self.cache_times[i] <= to_stamp]
```

Or even better you could even use `zip` to iterate over both lists at the same time.

**Comment by peci1** [comment]
I've implemented your suggestion with the constructor argument. I've also added tests. Should I retarget to kinetic-devel, or is it okay this way?

**Comment by tfoote** [comment]
lgtm, it would be better to target for kinetic and consider backporting only if there's strong demand

**Comment by peci1** [comment]
Ok, see #806 .

**Pull request closed without merging**